src/promptflow/promptflow/azure/_storage/cosmosdb/summary.py

```diff
@@ -0,0 +1,184 @@
+    line_number: str = None
+    # Only for line run
+    line_run_id: str = None
+
+
+class Summary:
+    __container__ = "LineSummary"
+
+    def __init__(self, span: Span) -> None:
+        self.span = span
+
+    def persist(self, client):
+        if self.span.parent_span_id:
+            # This is not the root span
+            return
+        attributes = self.span._content[SpanFieldName.ATTRIBUTES]
+
+        if (
+            SpanAttributeFieldName.LINE_RUN_ID not in attributes
+            and SpanAttributeFieldName.BATCH_RUN_ID not in attributes
+        ):
+            current_app.logger.info("No line run id or batch run id found. Could be aggregate node. Just ignore.")
+            return
+
+        # Persist span as a line run, since even evaluation is a line run, could be referenced by other evaluations.
+        self._persist_line_run(client)
+
+        if SpanAttributeFieldName.REFERENCED_LINE_RUN_ID in attributes or (
+            SpanAttributeFieldName.REFERENCED_BATCH_RUN_ID in attributes
+            and SpanAttributeFieldName.LINE_NUMBER in attributes
+        ):
+            self._insert_evaluation(client)
+
+    def _persist_line_run(self, client):
+        attributes: dict = self.span._content[SpanFieldName.ATTRIBUTES]
+
+        session_id = self.span.session_id
+        start_time = self.span._content[SpanFieldName.START_TIME]
+        end_time = self.span._content[SpanFieldName.END_TIME]
+
+        # Span's original format don't include latency, so we need to calculate it.
+        # Convert ISO 8601 formatted strings to datetime objects
+        start_time_date = datetime.datetime.fromisoformat(start_time.replace("Z", "+00:00"))
+        end_time_date = datetime.datetime.fromisoformat(end_time.replace("Z", "+00:00"))
+        latency = (end_time_date - start_time_date).total_seconds()
+        # calculate `cumulative_token_count`
+        completion_token_count = int(attributes.get(SpanAttributeFieldName.COMPLETION_TOKEN_COUNT, 0))
+        prompt_token_count = int(attributes.get(SpanAttributeFieldName.PROMPT_TOKEN_COUNT, 0))
+        total_token_count = int(attributes.get(SpanAttributeFieldName.TOTAL_TOKEN_COUNT, 0))
+        # if there is no token usage, set `cumulative_token_count` to None
+        if total_token_count > 0:
+            cumulative_token_count = {
+                "completion": completion_token_count,
+                "prompt": prompt_token_count,
+                "total": total_token_count,
+            }
+        else:
+            cumulative_token_count = None
+        item = SummaryLine(
+            id=self.span.trace_id,  # trace id is unique for LineSummary container
+            partition_key=session_id,
+            session_id=session_id,
+            trace_id=self.span.trace_id,
+            root_span_id=self.span.span_id,
+            inputs=json.loads(attributes[SpanAttributeFieldName.INPUTS]),
+            outputs=json.loads(attributes[SpanAttributeFieldName.OUTPUT]),
```

Comment:
what will happen if there is `float("nan")` in inputs/outputs? as they are not standard JSON value, and we have observed issues with those in local, and the solution is to hanlde in SDK side.

Reply:
Thanks a lot! For current code, cosmosDB will raise Exception for badrequest.